pxtlib/sprite-editor/canvasGrid.ts

```diff
@@ -333,34 +341,46 @@ namespace pxtsprite {
                 this.mouseRow
             ];
         }
+        protected drawFloatingLayer() {
+            if (!this.state.floatingLayer) {
+                return;
+            }
+            const context = this.paintLayer.getContext("2d");
+            this.drawImage(this.state.floatingLayer, context, this.state.layerOffsetX, this.state.layerOffsetY, true);
+
+            this.drawSelectionAnimation();
+        }
 
-        protected drawFloatingLayer(dashOffset = 0) {
+        protected drawSelectionAnimation(dashOffset = 0) {
             if (!this.state.floatingLayer) {
                 this.hideOverlay();
                 return;
             }
-
             this.showOverlay();
             const context = this.overlayLayer.getContext("2d");
-            context.clearRect(0, 0, this.overlayLayer.width, this.overlayLayer.height);
+            this.clearContext(context);
             context.globalAlpha = 1;
             context.strokeStyle = "#303030";
-            context.lineWidth = 5;
+            context.lineWidth = 2;
             context.setLineDash([5, 3]);
             context.lineDashOffset = dashOffset;
             context.strokeRect(this.state.layerOffsetX * this.cellWidth, this.state.layerOffsetY * this.cellHeight, this.state.floatingLayer.width * this.cellWidth, this.state.floatingLayer.height * this.cellHeight);
-            this.drawImage(this.state.floatingLayer, context, this.state.layerOffsetX, this.state.layerOffsetY, true);
+
 
             if (!this.lightMode && !this.selectAnimation && (!this.fadeAnimation || this.fadeAnimation.dead)) {
                 let drawLayer = () => {
                     dashOffset++
-                    requestAnimationFrame(() => this.drawFloatingLayer(dashOffset));
+                    requestAnimationFrame(() => this.drawSelectionAnimation(dashOffset));
                 };
 
                 this.selectAnimation = setInterval(drawLayer, 40)
             }
         }
 
+        private clearContext(context: CanvasRenderingContext2D) {
+            context.clearRect(0, 0, this.paintLayer.width, this.paintLayer.height);
```

Comment:
Add a comment that all three canvases are the same dimensions; (i.e. same as paintLayer)